internal/dnfjson/dnfjson.go

```diff
@@ -115,6 +115,10 @@ func (s *Solver) Depsolve(pkgSets []rpmmd.PackageSet) ([]rpmmd.PackageSpec, erro
 		return nil, err
 	}
 
+	// get non-exclusive read lock
+	s.cache.locker.RLock()
```

Comment:
I would squash this into the previous commit as it would make the consistency more obvious.

Reply:
(The whole commit I mean).

Reply:
Bit late for that now! :laughing: